app/bundles/EmailBundle/Model/SendEmailToContact.php

```diff
@@ -362,7 +362,7 @@ protected function processSendFailures($sendFailures)
             }
 
             // Add lead ID to list of failures
-            $this->failedContacts[$stat->getEmailId()] = $failedEmail;
+            $this->failedContacts[$stat->getLeadId()]  = $failedEmail;
```

Comment:
It is a bug. I think a code review is adequate. It's looping over contact email addresses but using the email ID as the key. This resulted in the $this->failedContacts array to always be the latest $failedEmail instead of collecting each of the failed addresses.  But to test, misconfigure momentum so that all sends fail. Send a broadcast/segment email to two contacts. Notice the UI will say "1 failed and 1 sent" and only list one email as failed. But the email was sent to neither.